workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs

```diff
@@ -136,6 +139,119 @@ private static void LaunchiOSDeviceClient()
             }
         }
 
+        [MenuItem(MenuLaunchiOSSimulator, false, 75)]
+        private static void LaunchiOSSimulatorClient()
+        {
+            try
+            {
+                EditorUtility.DisplayProgressBar("Launching iOS Simulator Client", "Launching iOS simulator", 0.2f);
+
+                // Open iOS simulator if it is not yet open
+                var xcodePath = GetXCodePath();
+                if (string.IsNullOrEmpty(xcodePath))
+                {
+                    Debug.LogError("Couldn't run xcode-select. Please make sure XCode is installed");
+                    return;
+                }
+
+                RedirectedProcess.Run("open", $"{xcodePath}/Applications/Simulator.app/");
+
+                EditorUtility.DisplayProgressBar("Launching iOS Simulator Client", "Installing client app", 0.5f);
+
+                // Find a built app to install
+                var bundleId = PlayerSettings.GetApplicationIdentifier(BuildTargetGroup.iOS);
+                var bundleName = bundleId.Split('.').LastOrDefault();
+                var appPath = Directory.GetDirectories(AbsoluteAppBuildPath, $"{bundleName}.app", SearchOption.AllDirectories).FirstOrDefault();
+                if (string.IsNullOrEmpty(appPath))
+                {
+                    Debug.LogError($"Could not find a built out iOS app in \"{AbsoluteAppBuildPath}\" to launch.");
+                    return;
+                }
+
+                // iOS simulator might not have finished starting at this point, so we want to give it some time
+                DateTime timeout = DateTime.Now.AddSeconds(30);
+                while (RedirectedProcess.Run("xcrun", $"simctl install booted {appPath}") != 0)
+                {
+                    if (DateTime.Now > timeout)
+                    {
+                        Debug.LogError("Error while installing app to the simulator. Please check the log for details about the error.");
+                    }
+
+                    System.Threading.Thread.Sleep(1000);
+                }
+
+                EditorUtility.DisplayProgressBar("Launching iOS Simulator Client", "Launching Client app", 0.8f);
+
+                // Optional arguments to be passed, same as standalone
+                // Use this to pass through the local ip to connect to
+                var runtimeIp = GdkToolsConfiguration.GetOrCreateInstance().RuntimeIp;
+                var arguments = new StringBuilder();
+                if (!string.IsNullOrEmpty(runtimeIp))
+                {
+                    arguments.Append($"+{RuntimeConfigNames.ReceptionistHost} {runtimeIp}");
+                }
+
+                var envVars = new Dictionary<string, string>
+                {
+                    { $"SIMCTL_CHILD_SPATIALOS_ARGUMENTS", arguments.ToString() }
+                };
+                if (RedirectedProcess.RunWithEnvVars(Path.GetFullPath(Path.Combine(Application.dataPath, "..")),
```

Comment:
The path should be in its own parameter

Reply:
Sorry, not sure I fully understand this - we do pass the path as a parameter of RunWithEnvVars parameter

Reply:
I mean that `Path.GetFullPath(Path.Combine(Application.dataPath, ".."))` should be pulled out into a seperate variable first.